beacon-chain/state/state-native/gloas.go

```diff
@@ -0,0 +1,78 @@
+package state_native
+
+import (
+	ethpb "github.com/OffchainLabs/prysm/v7/proto/prysm/v1alpha1"
+)
+
+// executionPayloadAvailabilityVal returns a copy of the execution payload availability.
+// This assumes that a lock is already held on BeaconState.
+func (b *BeaconState) executionPayloadAvailabilityVal() []byte {
+	if b.executionPayloadAvailability == nil {
+		return nil
+	}
+
+	availability := make([]byte, len(b.executionPayloadAvailability))
+	copy(availability, b.executionPayloadAvailability)
+
+	return availability
+}
+
+// builderPendingPaymentsVal returns a copy of the builder pending payments.
+// This assumes that a lock is already held on BeaconState.
+func (b *BeaconState) builderPendingPaymentsVal() []*ethpb.BuilderPendingPayment {
+	if b.builderPendingPayments == nil {
+		return nil
+	}
+
+	payments := make([]*ethpb.BuilderPendingPayment, len(b.builderPendingPayments))
+	for i, payment := range b.builderPendingPayments {
+		if payment != nil {
```

Comment:
You already have a nil check inside `Copy()` so this check is technically not needed